tests/Opus/Search/Common/IndexerTest.php

```diff
@@ -0,0 +1,961 @@
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testMissingConfigParamSearchEngine_Index_Path() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset(
+				$config->searchengine->solr->default->service->index->endpoint->primary->path,
+				$config->searchengine->solr->default->service->default->endpoint->primary->path
+			);
+
+			return $config;
+		} );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	public function testMissingConfigParamLogPrepareXml() {
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset( $config->log->prepare->xml );
+
+			return $config;
+		} );
+
+		$this->_addOneDocumentToIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testEmptyConfiguration() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(
+            'searchengine' => array( 'solr' => array( 'default' => array( 'service' => array( 'index' => array( 'endpoint' => array( 'primary' => array( 'path' => '' ) ) ) ) ) ) )
+        ) );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testInvalidConfiguration() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(
+            'searchengine' => array(
+                'solr' => array(
+                    'default' => array(
+                        'service' => array(
+                            'index' => array(
+	                            'endpoint' => array(
+		                            'primary' => array(
+			                            'host' => 'examplehost',
+			                            'port' => 'exampleport',
+			                            'path' => 'this_solr_instance_name_does_not_exist'
+		                            )
+	                            )
+                            )
+                        )
+                    )
+                )
+            )
+        ) );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	public function testPrepareAndOutputXML() {
+		$this->adjustConfiguration( array(
+			                            'log' => array( 'prepare' => array( 'xml' => true ) )
+		                            ) );
+
+		$this->_addOneDocumentToIndex();
+	}
+
+	public function testDeleteAllDocsInConstructor() {
```

Comment:
Delete. I didn't find any code using this behavior. If possible delete in separate commit.